pkg/build/strategies/sti/sti.go

```diff
@@ -567,7 +567,7 @@ func (b *STI) Execute(command string, user string, config *api.Config) error {
 			if err != nil {
 				// we're ignoring ErrClosedPipe, as this is information
 				// the docker container ended streaming logs
-				if glog.V(2) && err != io.ErrClosedPipe {
+				if glog.V(2) && err != io.ErrClosedPipe && err != io.EOF {
```

Comment:
no, not sure why it was done that way.  the new condition is because EOF is what we normally get now, possibly due to your goroutine fixes, I didn't consider it important enough to dig into.